internal/generator/vector/global.go

```diff
@@ -1,19 +1,22 @@
 package vector
 
 import (
+	"github.com/openshift/cluster-logging-operator/internal/collector/vector"
 	"github.com/openshift/cluster-logging-operator/internal/generator"
 )
 
-func Global() []generator.Element {
+func Global(name string) []generator.Element {
 	return []generator.Element{
 		GlobalOptions{
 			ExpireMetricsSecs: 60,
+			DataDir:           vector.DataPath + "/" + name,
```

Comment:
path.join or better would be to use expose a function from vector that provides the same value as what is used to generate the container spec